src/core/mainloop/periodic.c

```diff
@@ -307,12 +317,14 @@ periodic_events_rescan_by_roles(int roles, bool net_disabled)
   } SMARTLIST_FOREACH_END(item);
 }
 
-/** Invoked at shutdown: free resources used in this module.
+/**
+ * Invoked at shutdown: disconnect and unregister all periodic events.
  *
  * Does not free the periodic_event_item_t object themselves, because we do
- * not own them. */
+ * not own them.
+ */
 void
-periodic_events_destroy_all(void)
+periodic_events_disconnect_all(void)
```

Comment:
I do see the `periodic_event_destroy()` still a thing which is ONLY called by this `disconnect_all`. This is a bit confusing tbh since the function actually frees the event so "destroy" is valid. Since the `connect()` actually allocates the event, I think then we could safely rename the `_destroy()` to be `_disconnect()`.

And while at it, we might want to make it static since it is not used outside the module. Side note on that, all the specific `periodic_events_` seems only used within the module so if we want to narrow down the exposed API of `periodic.c`, we could static a bunch of those :).